avm/res/insights/data-collection-rule/tests/e2e/waf-aligned/main.test.bicep

```
@@ -0,0 +1,166 @@
+            '\\Process(_Total)\\Working Set'
+            '\\Process(_Total)\\Working Set - Private'
+            '\\LogicalDisk(_Total)\\% Disk Time'
+            '\\LogicalDisk(_Total)\\% Disk Read Time'
+            '\\LogicalDisk(_Total)\\% Disk Write Time'
+            '\\LogicalDisk(_Total)\\% Idle Time'
+            '\\LogicalDisk(_Total)\\Disk Bytes/sec'
+            '\\LogicalDisk(_Total)\\Disk Read Bytes/sec'
+            '\\LogicalDisk(_Total)\\Disk Write Bytes/sec'
+            '\\LogicalDisk(_Total)\\Disk Transfers/sec'
+            '\\LogicalDisk(_Total)\\Disk Reads/sec'
+            '\\LogicalDisk(_Total)\\Disk Writes/sec'
+            '\\LogicalDisk(_Total)\\Avg. Disk sec/Transfer'
+            '\\LogicalDisk(_Total)\\Avg. Disk sec/Read'
+            '\\LogicalDisk(_Total)\\Avg. Disk sec/Write'
+            '\\LogicalDisk(_Total)\\Avg. Disk Queue Length'
+            '\\LogicalDisk(_Total)\\Avg. Disk Read Queue Length'
+            '\\LogicalDisk(_Total)\\Avg. Disk Write Queue Length'
+            '\\LogicalDisk(_Total)\\% Free Space'
+            '\\LogicalDisk(_Total)\\Free Megabytes'
+            '\\Network Interface(*)\\Bytes Total/sec'
+            '\\Network Interface(*)\\Bytes Sent/sec'
+            '\\Network Interface(*)\\Bytes Received/sec'
+            '\\Network Interface(*)\\Packets/sec'
+            '\\Network Interface(*)\\Packets Sent/sec'
+            '\\Network Interface(*)\\Packets Received/sec'
+            '\\Network Interface(*)\\Packets Outbound Errors'
+            '\\Network Interface(*)\\Packets Received Errors'
+          ]
+        }
+      ]
+      windowsEventLogs: [
+        {
+          name: 'eventLogsDataSource'
+          streams: [
+            'Microsoft-Event'
+          ]
+          xPathQueries: [
+            'Application!*[System[(Level=1 or Level=2 or Level=3 or Level=4 or Level=0 or Level=5)]]'
+            'Security!*[System[(band(Keywords,13510798882111488))]]'
+            'System!*[System[(Level=1 or Level=2 or Level=3 or Level=4 or Level=0 or Level=5)]]'
+          ]
+        }
+      ]
+    }
+    destinations: {
+      azureMonitorMetrics: {
+        name: 'azureMonitorMetrics-default'
+      }
+      logAnalytics: [
+        {
+          workspaceResourceId: nestedDependencies.outputs.logAnalyticsWorkspaceResourceId
+          name: nestedDependencies.outputs.logAnalyticsWorkspaceName
+        }
+      ]
+    }
+    dataFlows: [
+      {
+        streams: [
+          'Microsoft-InsightsMetrics'
+        ]
+        destinations: [
+          'azureMonitorMetrics-default'
+        ]
+      }
+      {
+        streams: [
+          'Microsoft-Event'
+        ]
+        destinations: [
+          nestedDependencies.outputs.logAnalyticsWorkspaceName
+        ]
+      }
+
+    ]
+    kind: 'Windows'
+    tags: {
+      'hidden-title': 'This is visible in the resource name'
+      resourceType: 'Data Collection Rules'
+      kind: 'Windows'
+    }
```

```suggestion
    tags: {
      'hidden-title': 'This is visible in the resource name'
      resourceType: 'Data Collection Rules'
      kind: 'Windows'
    }
    lock: {
      kind: 'CanNotDelete'
      name: 'myCustomLockName'
    }
```
Please include lock in WAF Test

---

I'm also not sure @ChrisSidebotham - why the lock? As opposed to the e.g. `roleAssignments` I usually did not ask folks to remove 'locks' as I believe the former is more confusing to be there if you're interested in knowing how to deploy the service in a 'waf-aligned' way. I'd however say we definitely should ask folks to add locks :D

---

@AlexanderSehr not sure if I'm getting your comment :). At the beginning it sounds like "no locks", but the last sentence contradicts this 🙈. 
As for me, I'm paying attention that the lock deployment is included in one of the test cases (usually in "max"), but not necessarily in waf-aligned. But if you guys insist, i can add it here too.

---

For WAF Alignment I would add locks, aligning with the principles of reliability, security, and operational excellence implied in the framework supports the use of such practices. Resource locks protect workloads from unintended modifications or deletions, thus aligning with the framework's goals of ensuring reliability and security.